docs/antrea-network-policy.md

```diff
@@ -1334,6 +1335,96 @@ These two policies shown above are for demonstration purposes only. For more rea
 `namespaces` field, refer to this [sample](#acnp-for-strict-namespace-isolation) YAML in the previous
 section.
 
+### Selecting Namespaces of the same labels values with SameLabels
+
+Starting from Antrea v2.0, Antrea ClusterNetworkPolicy supports creating policy rules between groups
+of Namespaces based on values of some Namespace labels. The most preeminent use case of this feature
+is to provide isolation between Namespaces that have different values for some pre-defined labels,
+e.g. "org", by applying a single ACNP in the cluster.
+
+Consider a minimalistic cluster with the following Namespaces:
+
+```text
+NAME            LABELS
+kube-system     kubernetes.io/metadata.name=kube-system
+accounting1     kubernetes.io/metadata.name=accounting1, org=accounting, region=us-west
+accounting2     kubernetes.io/metadata.name=accounting2, org=accounting, region=us-east
+sales1          kubernetes.io/metadata.name=sales1, org=sales, region=us-west
+sales2          kubernetes.io/metadata.name=sales2, org=sales, region=us-east
+```
+
+An administrator of such cluster typically would want to enforce some boundaries between the "tenants"
+in the cluster (the accounting team and the sales team in this case, who each owns two Namespaces).
+This can be easily achieved by the following ACNP:
+
+```yaml
+apiVersion: crd.antrea.io/v1beta1
+kind: ClusterNetworkPolicy
+metadata:
+  name: isolation-based-on-org
+spec:
+  priority: 1
+  tier: securityops
+  appliedTo:
+    - namespaceSelector:
+        matchExpressions:
+          - { key: org, operator: Exists }
+  ingress:
+    - action: Allow
+      from:
+        - namespaces:
+            sameLabels: [org]
+    - action: Deny
+  egress:
+    - action: Allow
+      to:
+        - namespaces:
+            sameLabels: [org]
+    - action: Deny
+```
+
+The above policy can also adapt to the changes in the cluster, i.e. any new Namespace created in the
+cluster with a different "org" label value will be automatically isolated from both the accounting
+and the sales Namespaces. In addition, the Namespace grouping criteria can be easily extended to match
+more than one label keys, and Namespaces will be grouped together ONLY IF ALL the values of the label
+keys listed in the `sameLabels` field match. For example, if we change the `sameLabels` list to
+`[org, region]` in the example above, then this ACNP will create four Namespace groups instead of two,
+which are all isolated from each other. The reason is for the accounting or sales Namespaces, even though
+they share the same value for the 'org' label, they do have different values for the 'region' label.
```

Comment:
The reason is that individual Namespaces for the accounting or sales organizations have different values for the region label, even though they share the same value for the org label.